json_config_expander/config_expander.py

```diff
@@ -0,0 +1,49 @@
+"""
+ConfigExpander class makes all work
+"""
+from copy import deepcopy, copy
+from itertools import product, chain
+
+
+def seq_iter(obj):
+	# iterate dict and list in same way
+	return obj if isinstance(obj, dict) else range(len(obj))
+
+
+class ConfigExpander:
+	def __init__(self, expand_char='*'):
+		self.expand_char = expand_char
+
+	def run_on_each_config(self, base_config, function):
+		expanded_configs = self.expand_configs(base_config)
+		return [function(deepcopy(config)) for config in expanded_configs]
+
+	def expand_configs(self, base_config):
+		"""
+		For each child we calculate its versions recursively and create new config with all possible combinations
+		"""
+		if not isinstance(base_config, dict) and not isinstance(base_config, list):
+			return [base_config]
+		expanded_config = self._expand_config_children(base_config)
+		return self._create_cartesian_product_from_children(expanded_config)
+
+	def _expand_config_children(self, config):
+		expanded_config = copy(config)
```

Comment:
Just making sure - a flat copy here is inaf? Shouldn't it be a deepcopy? Btw, if you'll agree with me next comment this copy won't be necessary at all.